### PR TITLE
ManagedNetworkFabric - Updated CLI extension for API version 2023-06-15

### PR DESCRIPTION
**Pull request by rahulm-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
```
az networkfabric -h

Group
    az networkfabric : Manage Azure Network Fabric Management Service API.

Subgroups:
    acl                 : Manage Access Control List Resource.
    controller          : Manage Network Fabric Controller Resource.
    device              : Manage Network Device Resource.
    externalnetwork     : Manage External Network Resource.
    fabric              : Manage Network Fabric Resource.
    interface           : Manage Network Interface Resource.
    internalnetwork     : Manage Internal Network Resource.
    internetgateway     : Manage Internet Gateway Resource.
    internetgatewayrule : Manage Internet Gateway Rule Resource.
    ipcommunity         : Manage Ip Community Resource.
    ipextendedcommunity : Manage Ip Extended Community Resource.
    ipprefix            : Manage Ip Prefix Resource.
    l2domain            : Manage L2 Isolation Domain Resource.
    l3domain            : Manage L3 Isolation Domain Resource.
    nni                 : Manage Network To Network Interconnect Resource.
    rack                : Manage Network Rack Resource.
    routepolicy         : Manage Route Policy Resource.
```

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by kairu-ms — APPROVED** [review]

File: src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/acl/_create.py [hunk 1]
```diff
@@ -0,0 +1,1057 @@
...
+
+    :example: Help text for sub parameters under the specific parent can be viewed by using the shorthand syntax '??'. See https://github.com/Azure/azure-cli/tree/dev/doc/shorthand_syntax.md for more about shorthand syntax.
+        az networkfabric acl create --dynamic-match-configurations ??
```
Comment (kairu-ms) [inline src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/acl/_create.py:27]:
```suggestion
        az networkfabric acl create --dynamic-match-configurations "??"
```

File: src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/neighborgroup/_create.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    "networkfabric neighborgroup create",
+)
+class Create(AAZCommand):
+    """Create a Neighbor Group resource
+
+    :example: Create a Neighbor Group
+        az networkfabric neighborgroup create --resource-group "example-rg" --location "westus3" --resource-name "example-neighborgroup" --destination "{ipv4Addresses:['10.10.10.10']}"
+
+    :example: Help text for sub parameters under the specific parent can be viewed by using the shorthand syntax '??'. See https://github.com/Azure/azure-cli/tree/dev/doc/shorthand_syntax.md for more about shorthand syntax.
+        az networkfabric neighborgroup create --destination ??
```
Comment (kairu-ms) [inline src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/neighborgroup/_create.py:24]:
```suggestion
        az networkfabric neighborgroup create --destination "??"
```

File: src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/neighborgroup/_update.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    "networkfabric neighborgroup update",
+)
+class Update(AAZCommand):
+    """Update the Neighbor Group resource.
+
+    :example: Update the Neighbor Group
+        az networkfabric neighborgroup update --resource-group "example-rg" --resource-name "example-neighborgroup" --destination "{ipv4Addresses:['10.10.10.10'],ipv6Addresses:['2F::/100']}"
+
+    :example: Help text for sub parameters under the specific parent can be viewed by using the shorthand syntax '??'. See https://github.com/Azure/azure-cli/tree/dev/doc/shorthand_syntax.md for more about shorthand syntax.
+        az networkfabric neighborgroup update --destination ??
```
Comment (kairu-ms) [inline src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/neighborgroup/_update.py:24]:
```suggestion
        az networkfabric neighborgroup update --destination "??"
```

File: src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/routepolicy/_create.py [hunk 4]
```diff
@@ -18,10 +18,10 @@ class Create(AAZCommand):
...
+        az networkfabric routepolicy create --resource-group "example-rg" --resource-name "example-routepolicy" --location "westus3" --default-action "Permit" --nf-id "/subscriptions/xxxxx-xxxx-xxxx-xxxx-xxxxx/resourceGroups/example-rg/providers/Microsoft.ManagedNetworkFabric/networkFabrics/example-fabric" --address-family-type "IPv4" --statements "[{sequenceNumber:1235,condition:{ipExtendedCommunityIds:['/subscriptions/xxxxx-xxxx-xxxx-xxxx-xxxxx/resourceGroups/example-rg/providers/Microsoft.ManagedNetworkFabric/ipExtendedCommunities/example-ipExtendedCommunityName'],type:And},action:{localPreference:1235,actionType:Deny,ipExtendedCommunityProperties:{set:{ipExtendedCommunityIds:['/subscriptions/xxxxx-xxxx-xxxx-xxxx-xxxxx/resourceGroups/example-rg/providers/Microsoft.ManagedNetworkFabric/ipExtendedCommunities/example-ipExtendedCommunityName']}}}}]"
 
     :example: Help text for sub parameters under the specific parent can be viewed by using the shorthand syntax '??'. See https://github.com/Azure/azure-cli/tree/dev/doc/shorthand_syntax.md for more about shorthand syntax.
         az networkfabric routepolicy create --statements ??
```
Comment (kairu-ms) [inline src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/routepolicy/_create.py:27]:
```suggestion
        az networkfabric routepolicy create --statements "??"
```

File: src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/acl/_create.py (hunk 1 above)
Comment (rahulm-msft) [reply src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/acl/_create.py]: updated

File: src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/neighborgroup/_create.py (hunk 2 above)
Comment (rahulm-msft) [reply src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/neighborgroup/_create.py]: updated

File: src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/neighborgroup/_update.py (hunk 3 above)
Comment (rahulm-msft) [reply src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/neighborgroup/_update.py]: updated

File: src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/routepolicy/_create.py (hunk 4 above)
Comment (rahulm-msft) [reply src/managednetworkfabric/azext_managednetworkfabric/aaz/latest/networkfabric/routepolicy/_create.py]: updated

**Review by kairu-ms — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**